File: algorithms/11_calculator_menu.c
#include <stdio.h>
#include <stdlib.h>

void calculator(int operation);

int main()
{
    int operation;

    while (1)
    {
        printf("CALCULADORA - MENU\n");
        printf("1 - Adição\n");
        printf("2 - Subtração\n");
        printf("3 - Multiplicação\n");
        printf("4 - Divisão\n");
        printf("0 - Sair\n");

        scanf("%d", &operation);

        calculator(operation);
    }

    return 0;
}

void calculator(int operation)
{
    switch (operation)
    {
    case 1:
        printf("\n");
        for (int i = 0; i <= 9; i++)
        {
            for (int j = 0; j <= 9; j++)
            {
                printf("%d + %d = %d\n", i, j, (i + j));
            }
            printf("\n");
        }
        printf("\n");
        break;
    case 2:
        printf("\n");
        for (int i = 0; i <= 9; i++)
        {
            for (int j = 0; j <= 9; j++)
            {
                printf("%d - %d = %d\n", i, j, (i - j));
            }
            printf("\n");
        }
        printf("\n");
        break;
    case 3:
        printf("\n");
        for (int i = 0; i <= 9; i++)
        {
            for (int j = 0; j <= 9; j++)
            {
                printf("%d * %d = %d\n", i, j, (i * j));
            }
            printf("\n");
        }
        printf("\n");
        break;
    case 4:
        printf("\n");
        for (float i = 0; i <= 9; i++)
        {
            for (float j = 0; j <= 9; j++)
            {
                printf("%.0f / %.0f = %.0f\n", i, j, (i / j));
            }
            printf("\n");
        }
        printf("\n");
        break;
    case 0:
        exit(0);
        break;
    default:
        printf("Operação inválida\n");
        break;
    }
}

File: algorithms/11_calculator_menu.py

def calculator(operation):

    if operation == 1:
        print("\n")
        for i in range(10):
            for j in range(10):
                print(f"{i} + {j} = {i + j}")
            print("\n")
    elif operation == 2:
        print("\n")
        for i in range(10):
            for j in range(10):
                print(f"{i} - {j} = {i - j}")
            print("\n")
    elif operation == 3:
        print("\n")
        for i in range(10):
            for j in range(10):
                print(f"{i} * {j} = {i * j}")
            print("\n")
    elif operation == 4:
        print("\n")

        i = 1
        j = 1

        while(i <= 10):
            while(j <= 10):
                print(f"{i} / {j} = {i/j:.1f}")
                j += 1
                
            i += 1
            j = 1

        print("\n")
    elif operation == 0:
        exit()
    else:
        print("Operação inválida")


while(1):
    print("CALCULADORA - MENU")
    print("1 - Adição")
    print("2 - Subtração")
    print("3 - Multiplicação")
    print("4 - Divisão")
    print("0 - Sair")

    op = int(input("Qual a opção?\n"))

    calculator(op)
